nexus/external-api/src/lib.rs

```diff
@@ -0,0 +1,3005 @@
+use std::collections::BTreeMap;
+
+use anyhow::anyhow;
+use dropshot::{
+    EmptyScanParams, EndpointTagPolicy, HttpError, HttpResponseAccepted,
+    HttpResponseCreated, HttpResponseDeleted, HttpResponseFound,
+    HttpResponseHeaders, HttpResponseOk, HttpResponseSeeOther,
+    HttpResponseUpdatedNoContent, PaginationParams, Path, Query,
+    RequestContext, ResultsPage, StreamingBody, TypedBody,
+    WebsocketChannelResult, WebsocketConnection,
+};
+use http::Response;
+use hyper::Body;
+use ipnetwork::IpNetwork;
+use nexus_auth_types::authn::cookies::Cookies;
+use nexus_types::external_api::{params, shared, views};
+use omicron_common::api::external::{
+    http_pagination::{PaginatedById, PaginatedByName, PaginatedByNameOrId},
+    *,
+};
+use openapi_manager_types::ValidationContext;
+use openapiv3::OpenAPI;
+
+pub const API_VERSION: &str = "20240821.0";
+
+// API ENDPOINT FUNCTION NAMING CONVENTIONS
```

Comment:
Moved the function naming doc here because this is the canonical place for OpenAPI definitions.